include/mbedtls/asn1.h

```diff
@@ -240,7 +240,9 @@ int mbedtls_asn1_get_len(unsigned char **p,
  * \param p     On entry, \c *p points to the start of the ASN.1 element.
  *              On successful completion, \c *p points to the first byte
  *              after the length, i.e. the first byte of the content.
- *              On error, the value of \c *p is undefined.
+ *              If #MBEDTLS_ERR_ASN1_UNEXPECTED_TAG is returned, \c *p
```

Comment:
Can you give some examples of code that relies on this property? Does this extend to other functions as well (e.g. code that calls `mbedtls_asn1_get_int` and relies on not changing `*p` if there is a different tag?

Reply:
https://github.com/Mbed-TLS/mbedtls/blob/ab1f3c153a90d6943c174613b715d19dfddfc83a/library/x509_crt.c#L485-L493 is just one of _many_ examples.  If this behavior did not hold then parsing X.509 certificates without unique identifiers (which is the vast majority of them) would be broken.

Reply:
Thanks. Are you aware of indirect cases like assuming `mbedtls_asn1_get_int` doesn't change `*p`?

Reply:
https://github.com/Mbed-TLS/mbedtls/blob/6b4cb63aed23f16e1569a7a2bac0bcc8c0ed6afd/library/pkcs5.c#L89-L93 is the first one I found.